### PR TITLE
Add Real Enable/Disable Flags for Cross-Module Incremental Builds

### PR DESCRIPTION
**Pull request by CodaFi**
In the legacy driver, these flags will merely be propagated to the frontends to indicate that they should disable serialization of incremental information in swift module files.

In the new driver, these flags control whether the Swift driver performs an incremental build that is aware of metadata embedded in the module.

Kudos to David for coming up with our new marketing name: Incremental Imports.

For now, I am not removing the option to provide the old experimental flag. Once we re-run the options generation script in the new driver, I will be removing it.

rdar://74363450

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: include/swift/Option/Options.td [hunk 1]
```diff
@@ -1121,6 +1121,18 @@ def disable_request_based_incremental_dependencies : Flag<["-"],
...
+def enable_incremental_imports :
+  Flag<["-"], "enable-incremental-imports">,
+  Flags<[FrontendOption]>,
+  HelpText<"Enable cross-module incremental build metadata and "
+           "driver scheduling for Swift modules">;
+
+def disable_incremental_imports :
+  Flag<["-"], "disable-incremental-imports">,
+  Flags<[FrontendOption]>,
+  HelpText<"Enable cross-module incremental build metadata and "
```
Comment (CodaFi) [inline include/swift/Option/Options.td:1133]: Ack, thinko

**Review by davidungar — APPROVED** [review]
Looks good. Hoping this can land simultaneously with the Swift Driver change.

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: include/swift/Frontend/FrontendOptions.h [hunk 2]
```diff
@@ -298,7 +298,7 @@ class FrontendOptions {
   ///
   /// This flag is currently only propagated from the driver to
   /// any merge-modules jobs.
-  bool EnableExperimentalCrossModuleIncrementalBuild = false;
+  bool DisableCrossModuleIncrementalBuild = false;
```
[the remark this answers is not in the document]
Comment (CodaFi) [reply include/swift/Frontend/FrontendOptions.h:301]: It matches a the other enable/disable pairs we've got in the frontend is all. Generally, I'm with you.

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci test Windows

**Comment by CodaFi** [comment]
@swift-ci test Windows

**Comment by CodaFi** [comment]
@swift-ci smoke test macOS

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
⛵

**Pull request merged**